src/org/labkey/cds/CDSController.java

```diff
@@ -226,16 +226,33 @@ public void setUserProperties(JSONObject userProperties)
 
     private static final String ANALYTICS_USER_GROUP = "Active CAVD Member";
 
+    @RequiresNoPermission
+    public class UpdateNeedSurveyAction extends MutatingApiAction<Object>
+    {
+
+        @Override
+        public Object execute(Object o, BindException errors) throws Exception
+        {
+            CDSManager.get().setNeedSurvey(getUser(), true);
+
+            ApiSimpleResponse response  = new ApiSimpleResponse();
+            response.put("success", !errors.hasErrors());
+            return response;
+        }
+    }
+
     @RequiresNoPermission
     @IgnoresTermsOfUse
     public class AppAction extends SimpleViewAction
     {
         @Override
-        public ModelAndView getView(Object o, BindException errors) throws Exception
+        public ModelAndView getView(Object o, BindException errors)
         {
             Container c = getContainer();
             StudyService sss = StudyService.get();
             boolean hasStudy = null != sss && null != sss.getStudy(c);
+            boolean needSurvey = Boolean.parseBoolean(getViewContext().getActionURL().getParameter("NeedSurvey"));
```

Comment:
I don't think this is needed with the ajax, right?

Reply:
I had to add this specifically because  line 266, CDSManager.get().isNeedSurvey(getUser()) was returning false. Didn't think it was needed either, esp. since this code is executed as part of success callback after setting NeedSurvey to true.

Reply:
This is now fixed by OntologyManager to get NeedSurvey value as we discussed.